core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java

```diff
@@ -380,21 +360,16 @@ private void downloadResource() {
                 lock.notifyAll();
             }
             resource.fireDownloadEvent(); // fire ERROR
-        } finally {
-            if (connection != null) {
-                ConnectionFactory.getConnectionFactory().disconnect(connection);
-            }
         }
     }
 
-    private URLConnection getDownloadConnection(URL location) throws IOException {
-        URLConnection con = ConnectionFactory.getConnectionFactory().openConnection(location);
-        con.addRequestProperty("Accept-Encoding", "pack200-gzip, gzip");
-        con.connect();
-        return con;
+    private CloseableConnection getDownloadConnection(URL location) throws IOException {
```

Comment:
final param

Reply:
will do all the parameters in a later PR